### PR TITLE
FeatureFlags: Cleanup usage of cfg.IsFeatureToggleEnabled

### PR DESCRIPTION
**Pull request by ryantxu**
The original feature flag implementation was a function on settings `cfg.IsFeatureToggleEnabled` -- this was deprecated when FeatureFlags were added almost two years ago.

This PR refactors as much as possible to use `features.IsEnabled()` rather than the settings alternative.

Since the largest cascading use of cfg for settings is a sqlx toggle in the tag service, that was also removed

### CODE REVIEW

File: pkg/setting/setting.go [hunk 1]
```diff
@@ -145,6 +145,10 @@ var (
 	ImageUploadProvider string
 )
 
+type FeatureToggles interface {
+	IsEnabled(flag string) bool
+}
```
Comment (andresmgot) [inline pkg/setting/setting.go:150]: Can you use the same interface defined in pkg/services/featuremgmt/models.go?

**Review by IevaVasiljeva — APPROVED** [review]
LGTM for code owned by IAM.

File: pkg/setting/setting.go [hunk 2]
```diff
@@ -980,7 +984,24 @@ func NewCfg() *Cfg {
...
 		Azure:  &azsettings.AzureSettings{},
+
+		// Avoid nil pointer
+		IsFeatureToggleEnabled: func(_ string) bool {
+			return false
+		},
+	}
+}
+
+func NewCfgWithFeatures(features FeatureToggles) *Cfg {
```
Comment (andresmgot) [inline pkg/setting/setting.go:995]: Can you move this functionality to `NewCfg`? It would make this package easier to use.

File: pkg/setting/setting.go (hunk 1 above)
Comment (ryantxu) [reply pkg/setting/setting.go:153]: I added better comments and made it deprecated -- this is only here to avoid circular dependency with the new helper function

File: pkg/services/tag/tagimpl/tag.go [hunk 3]
```diff
@@ -4,23 +4,14 @@ import (
...
 	"github.com/grafana/grafana/pkg/services/tag"
-	"github.com/grafana/grafana/pkg/setting"
 )
 
 type Service struct {
 	store store
 }
 
-func ProvideService(db db.DB, cfg *setting.Cfg) *Service {
-	if cfg.IsFeatureToggleEnabled(featuremgmt.FlagNewDBLibrary) {
```
Comment (ryantxu) [inline pkg/services/tag/tagimpl/tag.go:17]: @zserge @papagian -- removing the sqlx tag implementation got added here since the cascade of changing most test files came from this 😬 

can split out to its own PR, but seems trivial enough

File: pkg/setting/setting.go (hunk 2 above)
Comment (andresmgot) [reply pkg/setting/setting.go]: Got it, thanks :+1:

**Review by andresmgot — APPROVED** [review]
LGTM, thanks for the comments.
